src/libstore/local-store.cc

```diff
@@ -559,21 +559,29 @@ void LocalStore::checkDerivationOutputs(const StorePath & drvPath, const Derivat
     };
 
 
-    if (drv.isFixedOutput()) {
-        DerivationOutputs::const_iterator out = drv.outputs.find("out");
-        if (out == drv.outputs.end())
-            throw Error("derivation '%s' does not have an output named 'out'", printStorePath(drvPath));
-
-        bool recursive; Hash h;
-        out->second.parseHashInfo(recursive, h);
-
-        check(makeFixedOutputPath(recursive, h, drvName), out->second.path, "out");
-    }
-
-    else {
-        Hash h = hashDerivationModulo(*this, drv, true);
-        for (auto & i : drv.outputs)
-            check(makeOutputPath(i.first, h, drvName), i.second.path, i.first);
+    // Don't need the answer, but do this anyways to assert is proper
+    // combination. The code below is more general and naturally allows
+    // combinations that are currently prohibited.
+    drv.type();
+
+    std::optional<Hash> h;
+    for (auto & i : drv.outputs) {
```

Comment:
I'm not sure what the point of this change is. Looks like it allows mixing FO and non-FO outputs in the same derivation, but that doesn't seem related to the rest of the changes :thinking: Or did I miss anything?

Reply:
Good catch, this is useful later for CA stuff, but doesn't need to be part of this PR.

Reply:
(I split it out of a branch that did more stuff for ca derivations, and I guess I forgot to remove this.)